src/webgl/light.js

```diff
@@ -494,6 +494,99 @@ p5.prototype.pointLight = function(v1, v2, v3, x, y, z) {
   return this;
 };
 
+/**
+ * Creates an image light with the given image.
+ *
+ * The image light stimulates light from all the directions.
+ * This is done by using the image as a texture for a relatively
+ * very huge light in a form of a sphere. This sphere contains
+ * or encapsulates the whole scene/drawing.
+ * It will have different effect for varying shininess of the
+ * object in the drawing.
+ * Under the hood it is mainly doing 2 types of calculations,
+ * first one is creating an irradiance map from the
+ * environment map of the input image.
+ * Second one is managing reflections based on the shininess
+ * or roughness of the material used in the scene.
+ *
+ * Note: The image's diffuse light will be affected by fill()
+ * and the specular reflections will be affected by specularMaterial()
+ * and shininess()
+ *
+ * @method imageLight
+ * @param  {p5.image}    img  image for the background
+ * @example
+ * <div class="notest">
+ * <code>
+ * let img;
+ * function preload() {
+ *   img = loadImage('assets/laDefense.jpg');
+ * }
+ * function setup() {
+ *   createCanvas(windowWidth, 400, WEBGL);
+ * }
+ * function draw() {
+ *   background(220);
+ *   imageMode(CENTER);
+ *   image(img, 0, 0, width, height);
+ *   ambientLight(50);
+ *   imageLight(img);
+ *   // This will use specular once we add it
+ *   specularMaterial(20);
+ *   // shininess(slider.value());
+ *   // orbitControl();
+ *   noStroke();
+ *   scale(2);
+ *   rotateX(frameCount * 0.005);
+ *   rotateY(frameCount * 0.005);
+ *   box(50);
+ * }
+ * </code>
+ * </div>
+ * @alt
+ * image light example
+ * @example
+ * <div class="notest">
+ * <code>
+ * let img;
+ * let slider;
+ * function preload() {
+ *   img = loadImage('assets/laDefense.jpg');
+ * }
+ * function setup() {
+ *   createCanvas(windowWidth, 400, WEBGL);
+ *   slider = createSlider(0, 800, 100, 1);
```

Comment:
I feel like I don't notice much change in visual after going halfway through the slider, so maybe for this example a range of 0-400 would be better?

Reply:
makes sense, done